pkg/cache/azure_cache.go

```diff
@@ -173,3 +172,12 @@ func (t *TimedCache) Set(key string, data interface{}) {
 		CreatedOn: time.Now().UTC(),
 	})
 }
+
+// Update updates the data cache for the key.
+func (t *TimedCache) Update(key string, data interface{}) {
```

Comment:
If we directly update the object, the object lock would be released automatically? see line 131

Reply:
Reverted the change as discussed offline.